File: frontend/src/api/paymentService.js
import axios from "axios";

const PAYMENT_SERVICE_URL = "http://localhost:8003";

export const makePayment = async (bookingId, amount) => {
    try {
        const response = await axios.post(`${PAYMENT_SERVICE_URL}/payments`, { bookingId, amount });
        return response.data;
    } catch (error) {
        console.error("Error making payment:", error);
        throw error;
    }
};

export const refundPayment = async (paymentId) => {
    try {
        const response = await axios.post(`${PAYMENT_SERVICE_URL}/payments/refund`, { paymentId });
        return response.data;
    } catch (error) {
        console.error("Error processing refund:", error);
        throw error;
    }
};

File: backend/payment-service/payment_service.py
from fastapi import FastAPI
from typing import Dict

app = FastAPI()

# Mock data for payment status
payments = {}

@app.post("/payment/{booking_id}",tags=["Payments"])
def process_payment(booking_id: int, amount: float):
    """
    **Process a payment for a flight booking.**
    
    - **flight_id**: ID of the booked flight
    - **amount**: Amount to be paid
    """
    # Assume successful payment
    payments[booking_id] = {"status": "Paid", "amount": amount}
    return {"message": f"Payment of ${amount} processed for booking {booking_id}"}

@app.post("/refund/{booking_id}",tags=["Refunds"])
def process_refund(booking_id: int):
    if booking_id in payments:
        payments[booking_id]["status"] = "Refunded"
        return {"message": f"Refund processed for booking {booking_id}"}
    return {"error": "Booking not found"}
